cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -127,34 +142,75 @@ def pad_index_two(index_string: str, reverse_complement: bool) -> str:
     return index_string + INDEX_TWO_PAD_SEQUENCE
 
 
-def adapt_indexes(
-    samples: List[FlowCellSample],
-    run_parameters: RunParameters,
+def get_hamming_distance_for_indexes(sequence_1: str, sequence_2: str) -> int:
+    """Get the hamming distance between two index sequences.
+    In the case that one sequence is longer than the other, the distance is calculated between
+    the shortest sequence and the first segment of equal length of the longest sequence."""
+    limit: int = min(len(sequence_1), len(sequence_2))
+    return get_hamming_distance(str_1=sequence_1[:limit], str_2=sequence_2[:limit])
+
+
+def update_barcode_mismatch_values_for_sample(
+    sample_to_update: FlowCellSampleNovaSeqX, samples: List[FlowCellSampleNovaSeqX]
 ) -> None:
-    """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
-    number  of index reads, and takes the reverse complement of index 2 in case of the new
-    novaseq software control version (1.7) in combination with the new reagent kit
-    (version 1.5).
-    """
-    LOG.info("Fix so that all indexes are in the correct format")
-    reverse_complement: bool = is_reverse_complement(run_parameters=run_parameters)
+    """Updates the sample's barcode mismatch values.
+    If a sample index has a hamming distance to any other sample lower than the threshold
+    (3 nucleotides), the barcode mismatch value for that index is set to zero."""
+    index_1_sample_to_update, index_2_sample_to_update = get_index_pair(sample=sample_to_update)
     for sample in samples:
-        index1, index2 = sample.index.split("-")
-        index1: str = index1.strip()
-        index2: str = index2.strip()
-        index_length = len(index1)
-        if is_padding_needed(
-            index_cycles=run_parameters.index_length, sample_index_length=index_length
+        if sample_to_update.sample_id == sample.sample_id:
+            continue
+        index_1, index_2 = get_index_pair(sample=sample)
+        if (
+            get_hamming_distance_for_indexes(
+                sequence_1=index_1_sample_to_update, sequence_2=index_1
+            )
+            < MINIMUM_HAMMING_DISTANCE
         ):
-            LOG.debug("Padding indexes")
-            index1 = pad_index_one(index_string=index1)
-            index2 = pad_index_two(index_string=index2, reverse_complement=reverse_complement)
-        if reverse_complement:
-            index2 = get_reverse_complement_dna_seq(index2)
-        sample.index = index1
-        sample.index2 = index2
+            LOG.debug(f"Turning barcode mismatch 1 to 0 for sample {sample_to_update.sample_id}")
+            sample_to_update.barcode_mismatches_1 = 0
```

Comment:
What is `sample_to_update.barcode_mismatches_1` before being set?

Reply:
The default value is 1, which is set when instantiating the FlowCellSample

Reply:
Does the 1 refer to the index? Maybe `barcode_mismatches_index_1` is a better name in that case